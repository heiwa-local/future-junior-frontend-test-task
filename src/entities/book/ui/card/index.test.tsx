import {render} from "@testing-library/react"
import { BookCard } from "./index"

describe("BookCard tests", () => {
    test("snapshot without data", () => {
        const bookCard = render(
            <BookCard
                id={undefined}
                imageUrl={undefined}
                title={undefined}
                authors={undefined}
                tag={undefined}
                onClick={() => {}}/>
        )
        expect(bookCard).toMatchSnapshot()
    })

    test("snapshot with data", () => {
        const bookCard = render(
            <BookCard
                id="z7nXDwAAQBAJ"
                imageUrl="http://books.google.com/books/publisher/content?id=z7nXDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&imgtk=AFLRE70InjG5KPOXyttqune1PgJTO5fvhSW-5PFUQCP_kTxCOsUxZ5AIjvg7rTLGi66zRoSVp9UmgukWFX6CbYygwQljhfqtLZ-Oi4SAHB_ofsY539j-9ntaePCGQEVeGiVCNBVhHROa&source=gbs_api"
                title="Prince Harry"
                authors={["Elizabeth Krajnik"]}
                tag="Young Adult Nonfiction"
                onClick={() => {}}/>
        )
        expect(bookCard).toMatchSnapshot()
    })
})