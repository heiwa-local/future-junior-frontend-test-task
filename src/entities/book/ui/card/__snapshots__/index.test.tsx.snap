// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard tests snapshot with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bookCard"
      >
        <div
          class="bookCard__imageWrap"
        >
          <div>
            <img
              class="bookCard__image"
              src="http://books.google.com/books/publisher/content?id=z7nXDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&imgtk=AFLRE70InjG5KPOXyttqune1PgJTO5fvhSW-5PFUQCP_kTxCOsUxZ5AIjvg7rTLGi66zRoSVp9UmgukWFX6CbYygwQljhfqtLZ-Oi4SAHB_ofsY539j-9ntaePCGQEVeGiVCNBVhHROa&source=gbs_api"
              style="display: none;"
            />
            <div
              class="bookCard__image"
              style="display: block;"
            >
              <span
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary loadingProgress css-18lrjg1-MuiCircularProgress-root"
                role="progressbar"
                style="width: 40px; height: 40px; margin: auto;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
            </div>
            <img
              class="bookCard__image"
              src="/not_found_image.png"
              style="display: none;"
            />
          </div>
        </div>
        <h3
          class="bookCard__title"
        >
          Prince Harry
        </h3>
        <h4
          class="bookCard__authors"
        >
          Elizabeth Krajnik
        </h4>
        <span
          class="bookCard__tag"
        >
          Young Adult Nonfiction
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bookCard"
    >
      <div
        class="bookCard__imageWrap"
      >
        <div>
          <img
            class="bookCard__image"
            src="http://books.google.com/books/publisher/content?id=z7nXDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&imgtk=AFLRE70InjG5KPOXyttqune1PgJTO5fvhSW-5PFUQCP_kTxCOsUxZ5AIjvg7rTLGi66zRoSVp9UmgukWFX6CbYygwQljhfqtLZ-Oi4SAHB_ofsY539j-9ntaePCGQEVeGiVCNBVhHROa&source=gbs_api"
            style="display: none;"
          />
          <div
            class="bookCard__image"
            style="display: block;"
          >
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary loadingProgress css-18lrjg1-MuiCircularProgress-root"
              role="progressbar"
              style="width: 40px; height: 40px; margin: auto;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </span>
          </div>
          <img
            class="bookCard__image"
            src="/not_found_image.png"
            style="display: none;"
          />
        </div>
      </div>
      <h3
        class="bookCard__title"
      >
        Prince Harry
      </h3>
      <h4
        class="bookCard__authors"
      >
        Elizabeth Krajnik
      </h4>
      <span
        class="bookCard__tag"
      >
        Young Adult Nonfiction
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BookCard tests snapshot without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bookCard"
      >
        <div
          class="bookCard__imageWrap"
        >
          <div>
            <img
              class="bookCard__image"
              style="display: none;"
            />
            <div
              class="bookCard__image"
              style="display: block;"
            >
              <span
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary loadingProgress css-18lrjg1-MuiCircularProgress-root"
                role="progressbar"
                style="width: 40px; height: 40px; margin: auto;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
            </div>
            <img
              class="bookCard__image"
              src="/not_found_image.png"
              style="display: none;"
            />
          </div>
        </div>
        <h3
          class="bookCard__title"
        />
        <h4
          class="bookCard__authors"
        />
        <span
          class="bookCard__tag"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bookCard"
    >
      <div
        class="bookCard__imageWrap"
      >
        <div>
          <img
            class="bookCard__image"
            style="display: none;"
          />
          <div
            class="bookCard__image"
            style="display: block;"
          >
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary loadingProgress css-18lrjg1-MuiCircularProgress-root"
              role="progressbar"
              style="width: 40px; height: 40px; margin: auto;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </span>
          </div>
          <img
            class="bookCard__image"
            src="/not_found_image.png"
            style="display: none;"
          />
        </div>
      </div>
      <h3
        class="bookCard__title"
      />
      <h4
        class="bookCard__authors"
      />
      <span
        class="bookCard__tag"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
